practical_07/project_management.py

```diff
@@ -0,0 +1,207 @@
+            except UnboundLocalError:
+                print("Projects must first be loaded")
+        elif menu_choice == "A":
+            try:
+                add_new_project(projects)
+            except UnboundLocalError:
+                print("Projects must first be loaded")
+        elif menu_choice == "U":
+            try:
+                update_project(projects)
+            except UnboundLocalError:
+                print("Projects must first be loaded")
+        else:
+            print("Invalid menu choice")
+        menu_choice = input(MENU).upper()
+
+
+def load_projects():
+    projects = []
+    filename = get_valid_string("Filename: ")
+    done = False
+    while not done:
+        try:
+            in_file = open(filename, 'r')
+            in_file.readline()  # Skip header
+            for line in in_file:
+                parts = line.strip().split('\t')
+                project = Project(str(parts[NAME_INDEX]),
+                                  (datetime.strptime(parts[START_DATE_INDEX], '%d/%m/%Y').date()),
+                                  int(parts[PRIORITY_INDEX]),
+                                  float(parts[COST_ESTIMATE_INDEX]), int(parts[COMPLETION_PERCENTAGE_INDEX]))
+                projects.append(project)
+            in_file.close()
+            print(f"Projects loaded")
+            done = True
+        except FileNotFoundError:
+            print("File not found")
+            filename = get_valid_string("Filename: ")
+    return projects
+
+
+def display_projects(projects):
+    print("Completed projects:")
+    projects.sort()  # Sort projects by priority
+    for project in projects:
+        if project.is_complete():
+            print(f"    {project}")
+    print("Uncompleted projects:")
+    for project in projects:
+        if not project.is_complete():
+            print(f"    {project}")
+
+
+def save_projects(projects):
+    filename = get_valid_string("Filename: ")
+    done = False
+    while not done:
+        try:
+            out_file = open(filename, 'w')
+            print("Name\tStart Date\tPriority\tCost Estimate\tCompletion Percentage", file=out_file)  # Print headers
+            for project in projects:
+                print(
+                    f"{project.name}\t{project.start_date.strftime('%d/%m/%Y')}\t{project.priority}\t{project.cost_estimate}\t{project.completion_percentage}",
+                    file=out_file)
+            out_file.close()
+            print(f"Projects saved in {filename}")
+            done = True
+        except FileExistsError:
+            print(f"File {filename} already exists")
+            filename = get_valid_string("Filename: ")
+
+
+def show_projects_after_date(projects):
+    input_date = get_valid_date("Show projects that begin after date (dd/mm/yyyy): ")
+    projects.sort(key=attrgetter('start_date'))
+    for project in projects:
+        if input_date <= project.start_date:
+            print(project)
```

Comment:
I had mentioned this in your project class, you could add another sorting method to the class that handles this?